nixos/modules/services/security/authentik.nix

```diff
@@ -0,0 +1,207 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib)
+    mkIf
+    types
+    mkOption
+    mkEnableOption
+    mkPackageOption
+    ;
+  cfg = config.services.authentik;
+
+  isRedisUnixSocket = lib.hasPrefix "/" cfg.redis.host;
+
+  commonServiceConfig = {
+    Type = "simple";
+    User = "authentik";
+    Group = "authentik";
+    Restart = "always";
+    RestartSec = 3;
+    RuntimeDirectory = "authentik";
+    NoNewPrivileges = true;
+    PrivateTmp = true;
+    ProtectHome = true;
+    ProtectSystem = "strict";
+    ProtectKernelTunables = true;
+    ProtectKernelModules = true;
+    ProtectControlGroups = true;
+    SystemCallFilter = "~@cpu-emulation @keyring @module @obsolete @raw-io @reboot @swap @sync";
+    ConfigurationDirectory = "authentik";
+  };
+in
+{
+  options.services.authentik = {
```

Comment:
Any benefit to adding support for running outposts as well?

Reply:
Long term yes, but I want to do this in a later PR

Reply:
Fair enough